plugins/performance-lab/includes/admin/plugins.php

```diff
@@ -101,12 +105,24 @@ function perflab_query_plugin_info( string $plugin_slug ) {
 		}
 	}
 
-	set_transient( $transient_key, $plugins, HOUR_IN_SECONDS );
-
 	if ( ! isset( $plugins[ $plugin_slug ] ) ) {
+		// Cache the fact that the plugin was not found.
+		$plugins[ $plugin_slug ] = array(
+			'error' => array(
+				'code'    => 'plugin_not_found',
+				'message' => __( 'Plugin not found in API response.', 'performance-lab' ),
+			),
+		);
+
+		set_transient( $transient_key, $plugins, MINUTE_IN_SECONDS );
+	} else {
+		set_transient( $transient_key, $plugins, HOUR_IN_SECONDS );
+	}
```

Comment:
This differentiation does not happen in the correct clause. We should use a shorter cache duration in case of any error, not just in the case of this specific error.

I think it would be most straightforward to implement this via a boolean flag like `$has_errors` that is set if anywhere in the (uncached) logic an error is set. Then the code here could be changed like:
```suggestion
	}

	if ( $has_errors ) {
		set_transient( $transient_key, $plugins, MINUTE_IN_SECONDS );
	} else {
		set_transient( $transient_key, $plugins, HOUR_IN_SECONDS );
	}
```